### PR TITLE
Feat/admin mailler

### PR DESCRIPTION
**Pull request by baron-trieu-goldenowl**
- Show list user
image

- Send mail subscription
image

- Add template send mail
image
image

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -3,5 +3,7 @@
 class HomeController < ApplicationController
   before_action :authenticate_user!
 
-  def index; end
+  def index
+    @user = User.all
```
Comment (techleadgo) [inline app/controllers/home_controller.rb:7]: Please apply pagination

File: app/models/career.rb [hunk 2]
```diff
@@ -32,4 +33,11 @@ class Career < ApplicationRecord
   def serializable_rich_content
     ActionController::Base.helpers.sanitize(ActionController::Base.helpers.raw(content))
   end
+
+  private
+
+  def send_mail
+    list_email = Subscription.where(subscription_type: 'career' || '').pluck(:email)
```
Comment (techleadgo) [inline app/models/career.rb:40]: Please test this query: `Subscription.where(subscription_type: 'career' || '')`. Does it include '' (blank) subscription_type?

File: app/models/career.rb [hunk 3]
```diff
@@ -32,4 +33,11 @@ class Career < ApplicationRecord
...
+    SubscriptionMailer.email_subscription(list_email).deliver_now
```
Comment (techleadgo) [inline app/models/career.rb:41]: What is the different between `deliver_now` and `deliver_later`?

File: app/views/home/index.html.slim [hunk 4]
```diff
@@ -9,10 +9,20 @@
...
+        .card-body
+          .toll-free-box.text-center
+            .media-content
+              h5.title.is-4 Admin Infomation
+              table.table.table-admin
+                tr
+                  th scope="col"  STT
+                  th scope="col"  ID
+                  th scope="col"  Email
+                - @user.each do |user|
```
Comment (techleadgo) [inline app/views/home/index.html.slim:24]: `@user` or `@users`? Please check the controller#index

**Comment by techleadgo** [comment]
Please screenshot mail content sent to users and put it in description too

File: config/environments/development.rb [hunk 5]
```diff
@@ -75,4 +75,16 @@
     Bullet.console = true
     Bullet.rails_logger = true
   end
+
+  # send mail subscription
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = {
+    user_name: 'AustinGO',
+    password: 'Golden2020',
+    domain: 'yourdomain.com',
```
Comment (techleadgo) [inline config/environments/development.rb:84]: Please use enviroment variables

File: config/environments/production.rb [hunk 6]
```diff
@@ -62,6 +62,9 @@
 
   config.action_mailer.perform_caching = false
 
+  # mailler default url
+  config.action_mailer.default_url_options = { host: 'http://eee.com' }
```
Comment (stewart-nguyen-goldenowl) [inline config/environments/production.rb:66]: Ask Austin for front-end url, use real host url at this config

File: app/models/subscription.rb [hunk 7]
```diff
@@ -12,4 +12,6 @@
 class Subscription < ApplicationRecord
   validates :name, presence: true
   validates :email, presence: true
+  scope :email_subscription_post, -> { where("subscription_type ISNULL or  subscription_type = '' or subscription_type = 'post'") }
```
Comment (stewart-nguyen-goldenowl) [inline app/models/subscription.rb:15]: Use `IS NULL` syntax instead of `ISNULL`

File: app/serializers/post_serializer.rb [hunk 8]
```diff
@@ -15,11 +15,14 @@
...
+  attributes  :title,
+              :created_at,
+              :updated_at
   attributes  :image do |post|
     post.image.url
   end
   attributes :content do |post|
     post.content.body.to_html
   end
+  attributes :source
```
Comment (stewart-nguyen-goldenowl) [inline app/serializers/post_serializer.rb:27]: place `:source` to line 18

**Pull request merged**